mlflow/utils/proto_json_utils.py

```diff
@@ -1,17 +1,81 @@
 import base64
 import datetime
 
+import json
 from json import JSONEncoder
 
 from google.protobuf.json_format import MessageToJson, ParseDict
+from google.protobuf.descriptor import FieldDescriptor
 
 from mlflow.exceptions import MlflowException
 from collections import defaultdict
+from functools import partial
+
+
+def _mark_int64_fields(proto_message):
+    """Converts a proto message to JSON, preserving only the int64/uint64/fixed64 fields."""
+
+    json_dict = {}
+    for field, value in proto_message.ListFields():
+        if field.type == FieldDescriptor.TYPE_MESSAGE:
+            ftype = partial(_mark_int64_fields)
+        elif field.type in [
+            FieldDescriptor.TYPE_INT64,
+            FieldDescriptor.TYPE_UINT64,
+            FieldDescriptor.TYPE_FIXED64,
+            FieldDescriptor.TYPE_SFIXED64,
+            FieldDescriptor.TYPE_SINT64,
+        ]:
+            ftype = int
+        else:
+            # Skip all non-int64 fields.
+            continue
+
+        json_dict[field.name] = (
+            [ftype(v) for v in value]
+            if field.label == FieldDescriptor.LABEL_REPEATED
+            else ftype(value)
+        )
+    return json_dict
+
+
+def _merge_json_dicts(from_dict, to_dict):
```

Comment:
Hi @lichenran1234, can we add a test for `_merge_json_dicts` to make sure it works properly?

Reply:
Thanks for the comment, Harupy! Usually we don't write unit tests for private functions. Especially for this `_merge_json_dicts` function: the code inside it should be embedded inside `message_to_json` function, but I'm extracting it out for readability. So I guess we should follow the [Test Behavior, Not Implementation](https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html) principle.

Can you think of more test cases for `message_to_json` if you are worried that `_merge_json_dicts` may not work properly?

Reply:
I agree with @lichenran1234 that don't test private function. But perhaps add more types of field to the unit test to verify that the public API works for all kinds of fields? e.g. we might want to verify that the following aspects are correctly translated:

- default values (I worry about this one)
- extensions (and this one)
- proto maps
- oneof
- enums

Reply:
@lichenran1234 Makes sense to not test `_merge_json_dicts`, thanks for the knowledge!

Reply:
Thanks @harupy and @jinzhang21 ! I added a new test proto message so I can test this function extensively. I also added support for proto maps.